### PR TITLE
Move hostscheduler code to pkg

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
- moves the hostscheduler test code to pkg to keep the cmd directories clean and have all business logic in `pkg``
- Only cleanup the cluster if the cluster is not yet freed
**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/testmachinery/argo/argo.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func getWorkflowAffinity() *corev1.Affinity {
 		NodeAffinity: &corev1.NodeAffinity{
 			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
 				{
-					Weight:     100,
+					Weight: 100,
```
Comment (dguendisch) [inline pkg/testmachinery/argo/argo.go]: Hmm, do we have different formatters?

Comment (schrodit) [reply pkg/testmachinery/argo/argo.go]: ah yeah could be. I use `go fmt`

Comment (dguendisch) [reply pkg/testmachinery/argo/argo.go]: I think goland is correct in this case. Could it be that you run an old version of gofmt? See e.g. https://github.com/golang/go/issues/26497

Comment (dguendisch) [reply pkg/testmachinery/argo/argo.go]: Well, just checked the latest gofmt (1.12.7) and it also wrongly formats the above...

Comment (dguendisch) [reply pkg/testmachinery/argo/argo.go]: Nevermind, then it is like it is and we stick to go fmt :)

**Pull request merged**